src/wallet/wallet2.cpp

```diff
@@ -1602,6 +1603,48 @@ std::string wallet2::get_subaddress_label(const cryptonote::subaddress_index& in
   return m_subaddress_labels[index.major][index.minor];
 }
 //----------------------------------------------------------------------------------------------------
+void wallet2::scan_tx(const std::set<std::string> txids)
```

Comment:
The copy seems pointless here. Also, wallet2 usually uses crypto::hash in its functions, simplewallet or the RPC server uses strings and converts.